pints/_optimisers/_irpropmin.py

```diff
@@ -178,7 +238,50 @@ def tell(self, reply):
         self._current_df = dfx
 
         # Take step in direction indicated by current gradient
-        self._proposed = self._current - self._step_size * np.sign(dfx)
+        p = self._current - self._step_size * np.sign(dfx)
+
+        # Allow boundaries to reduce step size
+        if self._lower is not None:
+            # Rectangular boundaries: reduce individual step sizes until OK
+            out = np.logical_or(p < self._lower, p >= self._upper)
```

Comment:
Is there a reason for including the upper boundary and not the lower boundary?

Reply:
Yeah, the RectangularBoundaries class uses a half-open interval: https://github.com/pints-team/pints/blob/master/pints/_boundaries.py#L91-L97

This code is doing the same check manually, so should use the same convention